taca_ngi_pipeline/deliver/deliver_castor.py

```diff
@@ -0,0 +1,181 @@
+        self.castorsftpserver = getattr(self, 'castorsftpserver', None)
+        self.castorsftpserver_user = getattr(self, 'castorsftpserver_user', None)
+    
+    def create_sftp_connnection(self):
+        try:
+            self.transport=paramiko.Transport(self.castorsftpserver)
+            password = getpass.getpass(prompt='Bianca/Castor Password for user {}:'.format(self.castorsftpserver_user))
+            self.transport.connect(username = "{}-{}".format(self.castorsftpserver_user, self.uppnexid), password = password)
+        except Exception as e:
+            logger.error("Caught exception: {}: {}".format(e.__class__, e))
+            raise
+        #open the sftp client
+        self.sftp_client = MySFTPClient.from_transport(self.transport)
+
+    def close_sftp_connnection(self):
+        try:
+            #now I can close the client
+            self.sftp_client.close()
+            #and the transport
+            self.transport.close()
+        except Exception as e:
+            logger.error("Caught exception: {}: {}".format(e.__class__, e))
+            raise
+
+    def deliver_project(self):
+        """ Deliver all samples in a project to castor
+            
+            :returns: True if all samples were delivered successfully, False if
+                any sample was not properly delivered or ready to be delivered
+        """
+        try:
+            logger.info("Delivering {} to {}".format(
+                str(self), self.expand_path(self.castordeliverypath)))
+            if self.get_delivery_status() == 'DELIVERED' \
+                    and not self.force:
+                logger.info("{} has already been delivered".format(str(self)))
+                return True
+            status = True
+            #open one client session and leave it open for all the time of the transfer
+            self.create_sftp_connnection()
+            #memorize all samples that needs to be delivered
+            samples_to_deliver = [sentry['sampleid'] for sentry in db.project_sample_entries(
+                    db.dbcon(), self.projectid).get('samples', [])]
+            status = True
+            # move to the delivery directory in the sftp
+            self.sftp_client.chdir(self.expand_path(self.castordeliverypath))
+            #create the project folder in the remote server
+            self.sftp_client.mkdir(self.projectid, ignore_existing=True)
+            #move inside the project folder
+            self.sftp_client.chdir(self.projectid)
+            #now cycle across the samples
+            for sampleid in samples_to_deliver:
+                sampleDelivererObj = CastorSampleDeliverer(self.projectid, sampleid, self.sftp_client)
+                st = sampleDelivererObj.deliver_sample()
+                status = (status and st)
+            # query the database whether all samples in the project have been sucessfully delivered
+            if self.all_samples_delivered():
```

Comment:
No good catch!!! I changed how I handle ABORTED by setting them to DELIVERED... not very clean but otherwise I need to change the way in which the DB is queried.